### ISSUE TITLE
Generated constructor do not have annotations

### ISSUE THREAD

**Issue by tasomaniac** [rharter/auto-value-gson#147]
When properties have annotations, `AutoValue` puts their respective annotations to the parameters in the constructor. This library does not, this causes nullity problems with some tools. It is a problem with kotlin too. 

Here as an example the generated code from `AutoValue` and this library respestively. 

```java
$AutoValue_Section(
      @Nullable String title,
      @Nullable String featuredBarColor,
      @Nullable String featuredRealtorLogoUrl)
```

```java
  AutoValue_Section(String title, String featuredBarColor,
      String featuredRealtorLogoUrl) {
    super(title, featuredBarColor, featuredRealtorLogoUrl);
  }
```

**Comment by JakeWharton** [comment]
Fixed by #144

**Comment by tasomaniac** [comment]
I checked the issues but couldn't find it. Sorry for duplicate issue. 👍
